narwhal/consensus/src/consensus.rs

```diff
@@ -31,6 +35,114 @@ pub mod consensus_tests;
 /// The representation of the DAG in memory.
 pub type Dag = BTreeMap<Round, HashMap<AuthorityIdentifier, (CertificateDigest, Certificate)>>;
 
+#[derive(Default)]
+pub struct LeaderSwapTable {
```

Comment:
feels like the consensus.rs is getting a bit bloated. Would it be a good idea to create a node reputation and leader election module?

Reply:
Good idea! Let me open a few more PRs with more scope and can evaluate how things could be refactored once have a better idea of the final shape here.